### ISSUE TITLE
It supports only for 10-20 secs

### ISSUE THREAD

**Issue by AyyazAnjum** [Rapsssito/react-native-background-actions#235]
When my app is in background or is killed from background my function runs forruns for only short period of time like 10 to 20 secs only

**Comment by MalekZishan** [comment]
Yes same i am facing now

**Comment by rajkumarthirumalai** [comment]
yes facing the same even manually changeing service type to dataSync it works properly for 14 but not on lower device

**Comment by AyyazAnjum** [comment]
> yes facing the same even manually changing service type to dataSync it works properly for android 14 version but not on android  lower version device
> 

What your code is working for latest Android and not for lower devices
How?
I'm facing the same issue but vice versa 
Working on lower devices but not on lastest

**Comment by rajkumarthirumalai** [comment]
For this you have to change the  nodemodules 
node_modules/react-native-background-actions/android/src/main/AndroidManifest.xml 

EXISTING CODE
<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.asterinet.react.bgactions">
    <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
    <application>
        <service android:name=".RNBackgroundActionsTask"/>
    </application>
</manifest>

UPDATED ONE
<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.asterinet.react.bgactions">
    <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
    <application>
        <service android:name=".RNBackgroundActionsTask" android:foregroundServiceType="dataSync" />
    </application>
</manifest>

**Comment by deban07** [comment]
> When my app is in background or is killed from background my function runs for only short period of time like 10 to 20 secs only for android 12+ but for android below 12+ it is working fine...

     It will work for all android devices .

     Add these at   =>   android/app/src/main/AndroidManifest.xml

    <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_DATA_SYNC" />


   <service android:name="com.supersami.foregroundservice.ForegroundService"></service>
  <service android:name="com.supersami.foregroundservice.ForegroundServiceTask"></service>
  <service android:name="com.asterinet.react.bgactions.RNBackgroundActionsTask" android:foregroundServiceType="dataSync"/>

**Comment by anurastogi** [comment]
The answer by deban07 and rajkumarthirumalai worked perfectly with version 4.0.1

**Comment by vishalyad16** [comment]
@deban07 @AyyazAnjum @deban07 @rajkumarthirumalai @MalekZishan still crashing on my side

**Comment by zouinekh** [comment]
> > When my app is in background or is killed from background my function runs for only short period of time like 10 to 20 secs only for android 12+ but for android below 12+ it is working fine...
> 
> ```
>  It will work for all android devices .
> 
>  Add these at   =>   android/app/src/main/AndroidManifest.xml
> 
> <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
> <uses-permission android:name="android.permission.FOREGROUND_SERVICE_DATA_SYNC" />
> ```

wlh mak mnayk sa7by it works 3alkher

**Comment by medkhalilbk** [comment]
> > When my app is in background or is killed from background my function runs for only short period of time like 10 to 20 secs only for android 12+ but for android below 12+ it is working fine...
> 
> ```
>  It will work for all android devices .
> 
>  Add these at   =>   android/app/src/main/AndroidManifest.xml
> 
> <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
> <uses-permission android:name="android.permission.FOREGROUND_SERVICE_DATA_SYNC" />
> ```

Thanks [deban07](https://github.com/deban07) you saved my life!

**Comment by deban07** [comment]
> @deban07 @AyyazAnjum @deban07 @rajkumarthirumalai @MalekZishan still crashing on my side

Which android version you are using and mention the React Native version also.

**Comment by rajkumarthirumalai** [comment]
@deban07 i have stopped using this package usage and used native foreground service from android 

1.create a java serviceFile 
`package com.yourpackage.service;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.IBinder;
import androidx.core.app.NotificationCompat;

public class BackgroundService extends Service {
    private static final int NOTIFICATION_ID = 1;
    private static final String CHANNEL_ID = "BackgroundServiceChannel";

    @Override
    public void onCreate() {
        super.onCreate();
        createNotificationChannel();
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        startForeground(NOTIFICATION_ID, createNotification());
        // Your long-running task goes here
        return START_STICKY;
    }

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    private void createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel serviceChannel = new NotificationChannel(
                CHANNEL_ID,
                "Background Service Channel",
                NotificationManager.IMPORTANCE_DEFAULT
            );
            NotificationManager manager = getSystemService(NotificationManager.class);
            manager.createNotificationChannel(serviceChannel);
        }
    }

    private Notification createNotification() {
        NotificationCompat.Builder builder = new NotificationCompat.Builder(this, CHANNEL_ID)
            .setContentTitle("Background Service")
            .setContentText("Service is running in the background")
            .setSmallIcon(R.mipmap.ic_launcher)
            .setPriority(NotificationCompat.PRIORITY_DEFAULT);

        return builder.build();
    }
}`
2.Add this manifest

`<service
            android:name=".BackgroundService"
            android:foregroundServiceType="dataSync"
            android:enabled="true"
            android:exported="false" />`


3.now comes the module 
`package com.yourpackage.module;

import android.content.Intent;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.yourpackage.service.BackgroundService;

public class BackgroundServiceModule extends ReactContextBaseJavaModule {
    private final ReactApplicationContext reactContext;

    public BackgroundServiceModule(ReactApplicationContext reactContext) {
        super(reactContext);
        this.reactContext = reactContext;
    }

    @Override
    public String getName() {
        return "BackgroundService";
    }

    @ReactMethod
    public void startService() {
        Intent serviceIntent = new Intent(reactContext, BackgroundService.class);
        reactContext.startService(serviceIntent);
    }

    @ReactMethod
    public void stopService() {
        Intent serviceIntent = new Intent(reactContext, BackgroundService.class);
        reactContext.stopService(serviceIntent);
    }
}`

4. need to add the package to module
`package com.yourpackage;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;
import com.yourpackage.module.BackgroundServiceModule;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class BackgroundServicePackage implements ReactPackage {
    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }

    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        List<NativeModule> modules = new ArrayList<>();
        modules.add(new BackgroundServiceModule(reactContext));
        return modules;
    }
}`

5.Finally at our jsx :
`import { NativeModules } from 'react-native';

const { BackgroundService } = NativeModules;

// Start the service
BackgroundService.startService();

// Stop the service
BackgroundService.stopService();`

**Comment by Ezzine-Smichi** [comment]
i am facing this bug now in the ios platform, there is a solution for this or not yet
